### PR TITLE
SDI-549 Update to key date adjustment

### PR DESCRIPTION
**Pull request by andymarke**
also made language consistent with adjustment IDs

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ import java.time.LocalDate
 @RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
 class SentencingAdjustmentResource(private val sentencingAdjustmentService: SentencingAdjustmentService) {
   @PreAuthorize("hasRole('ROLE_NOMIS_SENTENCING')")
-  @GetMapping("/sentence-adjustments/{sentenceAdjustmentId}")
+  @GetMapping("/sentence-adjustments/{adjustmentId}")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt:36]: minor language change

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentService.kt [hunk 2]
```diff
@@ -129,8 +129,28 @@ class SentencingAdjustmentService(
...
+      entityManager.flush()
+      storedProcedureRepository.postKeyDateAdjustmentCreation(
+        keyDateAdjustmentId = adjustmentId,
+        bookingId = this.offenderBooking.bookingId
+      )
+    } ?: throw NotFoundException("Key date adjustment with id $adjustmentId not found")
+
   fun findAdjustmentIdsByFilter(pageRequest: Pageable, adjustmentFilter: AdjustmentFilter): Page<AdjustmentIdResponse> {
     val adjustedToDate = adjustmentFilter.toDate?.let { adjustmentFilter.toDate.plusDays(1) }
-    return keyDateAdjustmentRepository.adjustmentIdsQuery_named(fromDate = adjustmentFilter.fromDate, toDate = adjustedToDate, pageRequest)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentService.kt:134]: formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt [hunk 3]
```diff
@@ -215,8 +215,47 @@ class SentencingAdjustmentResource(private val sentencingAdjustmentService: Sent
...
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "403",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt:242]: No 404?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt (hunk 3 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt:242]: yep, missed that, next PR

**Pull request merged**